src/server.rs

```diff
@@ -409,47 +410,21 @@ fn convert_public_key(native_pub_key: PublicKey) -> lndkrpc::PublicKey {
     lndkrpc::PublicKey { key: pub_key_bytes }
 }
 
-// Conversion function for FeatureBit.
-// TODO: Converting the FeatureBits doesn't work quite properly right now.
-fn feature_bit_from_id(feature_id: u8) -> Option<FeatureBit> {
-    match feature_id {
-        0 => Some(FeatureBit::DatalossProtectOpt),
-        1 => Some(FeatureBit::DatalossProtectOpt),
-        3 => Some(FeatureBit::InitialRouingSync),
-        4 => Some(FeatureBit::UpfrontShutdownScriptReq),
-        5 => Some(FeatureBit::UpfrontShutdownScriptOpt),
-        6 => Some(FeatureBit::GossipQueriesReq),
-        7 => Some(FeatureBit::GossipQueriesOpt),
-        8 => Some(FeatureBit::TlvOnionReq),
-        9 => Some(FeatureBit::TlvOnionOpt),
-        10 => Some(FeatureBit::ExtGossipQueriesReq),
-        11 => Some(FeatureBit::ExtGossipQueriesOpt),
-        12 => Some(FeatureBit::StaticRemoteKeyReq),
-        13 => Some(FeatureBit::StaticRemoteKeyOpt),
-        14 => Some(FeatureBit::PaymentAddrReq),
-        15 => Some(FeatureBit::PaymentAddrOpt),
-        16 => Some(FeatureBit::MppReq),
-        17 => Some(FeatureBit::MppOpt),
-        18 => Some(FeatureBit::WumboChannelsReq),
-        19 => Some(FeatureBit::WumboChannelsOpt),
-        20 => Some(FeatureBit::AnchorsReq),
-        21 => Some(FeatureBit::AnchorsOpt),
-        22 => Some(FeatureBit::AnchorsZeroFeeHtlcReq),
-        23 => Some(FeatureBit::AnchorsZeroFeeHtlcOpt),
-        30 => Some(FeatureBit::AmpReq),
-        31 => Some(FeatureBit::AmpOpt),
-        _ => None,
+// Conversion function for invoice features.
+fn convert_invoice_features(features: Bolt12InvoiceFeatures) -> Vec<i32> {
+    let mut feature_bits = Vec::new();
+    if features.requires_basic_mpp() {
+        feature_bits.push(FeatureBit::MppReq as i32);
     }
+    if features.supports_basic_mpp() {
+        feature_bits.push(FeatureBit::MppOpt as i32);
+    }
+    feature_bits
 }
 
-// Conversion function for features.
-// TODO: Converting the FeatureBits doesn't work quite properly right now.
-fn convert_features(features: Vec<u8>) -> Vec<i32> {
-    features
-        .iter()
-        .filter_map(|&feature_id| feature_bit_from_id(feature_id))
-        .map(|feature_bit| feature_bit as i32) // Cast enum variant to i32
-        .collect()
+// Conversion function for hop features.
+fn convert_hop_features(_: BlindedHopFeatures) -> Vec<i32> {
+    Vec::new()
```

Comment:
Hmm is this supposed to just return an empty vector?

Reply:
According to https://docs.rs/lightning/latest/src/lightning/ln/features.rs.html#202-210, `BlindedHopContext` has no possible features.

Reply:
Ah I see. Could you just add a comment to the function explaining + the link? And a similar comment to `convert_invoice_features` - just in case changes are made to those features in the future, it'd be good to keep track

Reply:
Pushed updated commit in https://github.com/lndk-org/lndk/pull/145/commits/0eb7d7e8738de07a9d0c36a8b246c803ddc9d4e4 with reference to rust-lightning source code off which this is based.